package/base-files/files/lib/upgrade/common.sh

```diff
@@ -231,11 +231,17 @@ indicate_upgrade() {
 # $(1): path to image
 # $(2): (optional) pipe command to extract firmware, e.g. dd bs=n skip=m
 default_do_upgrade() {
+	local err
 	sync
 	if [ -n "$UPGRADE_BACKUP" ]; then
 		get_image "$1" "$2" | mtd $MTD_ARGS $MTD_CONFIG_ARGS -j "$UPGRADE_BACKUP" write - "${PART_NAME:-image}"
 	else
 		get_image "$1" "$2" | mtd $MTD_ARGS write - "${PART_NAME:-image}"
 	fi
-	[ $? -ne 0 ] && exit 1
+	err=$?
+	if [ $err -ne 0 ] ; then
+		>&2 echo "ERROR: default_do_upgrade mtd write error: $err"
```

Comment:
Will be changed in next push

Reply:
Ah...I'm not applying this change due to the reason explained in the following comment: This commit itself doesn't serve the purpose of "return the error for caller to deal with" because it can only return with a success or terminate sysupgrade entirely.

Reply:
Will remove the commit in next push